Queue/Consumer.php

```diff
@@ -0,0 +1,181 @@
+ */
+class Consumer implements ConsumerInterface
+{
+    public function __construct(
+        private CallbackInvokerInterface $invoker,
+        private MessageEncoder $messageEncoder,
+        private ConsumerConfigurationInterface $configuration,
+        private LoggerInterface $logger,
+        private ConsumerConfig $consumerConfig,
+        private CommunicationConfig $communicationConfig,
+        private QueueRepository $queueRepository,
+        private MessageController $messageController,
+        private MessageValidator $messageValidator,
+        private EnvelopeFactory $envelopeFactory,
+        private QueueLockRepository $queueLockRepository,
+        private HandleQueueFailureService $handleQueueFailureService
+    ) {
+    }
+
+    public function process($maxNumberOfMessages = null): void
+    {
+        $queue = $this->configuration->getQueue();
+        $maxIdleTime = $this->configuration->getMaxIdleTime();
+        $sleep = $this->configuration->getSleep();
+
+        if (!isset($maxNumberOfMessages)) {
+            $queue->subscribe($this->getTransactionCallback($queue));
+        } else {
+            $this->invoker->invoke(
+                $queue,
+                $maxNumberOfMessages,
+                $this->getTransactionCallback($queue),
+                $maxIdleTime,
+                $sleep
+            );
+        }
+    }
+
+    private function getTransactionCallback(QueueInterface $queue): \Closure
+    {
+        return function (EnvelopeInterface $message) use ($queue) {
+            /** @var LockInterface $lock */
+            $lock = null;
+
+            try {
+                $topicName = $message->getProperties()['topic_name'];
+                $topicConfig = $this->communicationConfig->getTopic($topicName);
+                $lock = $this->messageController->lock($message, $this->configuration->getConsumerName());
+
+                if ($topicConfig[CommunicationConfig::TOPIC_IS_SYNCHRONOUS]) {
+                    $responseBody = $this->dispatchMessage($message, true);
+                    $responseMessage = $this->envelopeFactory->create(
+                        [ 'body' => $responseBody, 'properties' => $message->getProperties() ]
+                    );
+                    $this->sendResponse($responseMessage);
+                } else {
+                    $allowedTopics = $this->configuration->getTopicNames();
+
+                    if (!in_array($topicName, $allowedTopics)) {
+                        $queue->reject($message);
+                        return;
+                    }
+
+                    $this->dispatchMessage($message);
+                }
+
+                $queue->acknowledge($message);
+            } catch (MessageLockException $exception) {
+                $queue->acknowledge($message);
+            } catch (ConnectionLostException $exception) {
+                if ($lock) {
+                    $this->queueLockRepository->deleteById((int)$lock->getId());
```

Comment:
This line was introduced to use that `queueLockRepository` instead of the resource model